source/detail/time.hpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2024 xlnt-community
```

Comment:
I was honestly a bit unsure how to call it - whether it should be a general file for OS-specific wrappers, or something time-specific. However, to keep the number of includes to a minimum and to group the functions in a better way, I decided to create a file for time-specific functions (date, time, date/time) and simply called it `time.hpp`. But I'm open for suggestions :wink: We could also call it something like `time_wrappers.hpp` - on the other hand, all the other filenames in the same directory are kept rather short.

Reply:
I'm just wondering if the same filename could give some trouble on certain platforms. If not, we can keep time.hpp, otherwise time_wrappers, or more general time_helpers or time_conversion may be better.

Reply:
I really do hope that no compiler and/or build system would work so badly that different paths containing the same filename would cause issues :sweat_smile: What I would rather think about it is that it would be easier for other contributors to XLNT to open the wrong file, causing confusions. This is however already an issue with many other libraries like the C++ STL and Boost (they often have similar filenames), so we are not the first ones having this issue. This is more of a theoretical nature, as I'm not aware of people actually complaining about this. But if you prefer me to change the name, I'll do it, of course :wink:

Reply:
Batch/inference rules in NMake may be an example (at least if generated by qmake for source files): see https://bugreports.qt.io/browse/QTVSADDINBUG-424, but that's maybe irrelevant for our use case. So, leave it as you like.

Reply:
Alright, I renamed it to `time_helpers.hpp` :+1: